tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/ModifyRepositoryMetadataMojo.java

```diff
@@ -0,0 +1,158 @@
+     */
+    @Parameter(property = "p2.repository.properties.remove")
+    private List<String> propertiesToRemove = List.of();
+    /**
+     * The properties to add.<br>
+     * Already existing properties are overwritten and updated to the value specified.
+     */
+    @Parameter(property = "p2.repository.properties.add")
+    private Map<String, String> propertiesToAdd = Map.of();
+
+    /** Compress the repository index files {@code content.xml} and {@code artifacts.xml}. */
+    @Parameter(property = "p2.repository.compress", defaultValue = "true")
+    private boolean compress;
+
+    /**
+     * Add XZ-compressed repository index files. XZ offers better compression ratios esp. for highly
+     * redundant file content.
+     */
+    @Parameter(property = "p2.repository.xz", defaultValue = "true")
+    private boolean xzCompress;
+
+    /**
+     * If {@link #xzCompress} is {@code true}, whether jar or xml index files should be kept in
+     * addition to XZ-compressed index files. This fallback provides backwards compatibility for
+     * pre-Mars p2 clients which cannot read XZ-compressed index files.
+     */
+    @Parameter(property = "p2.repository.xz.keep", defaultValue = "true")
+    private boolean keepNonXzIndexFiles;
+
+    @Inject
+    private P2RepositoryDataManipulator manipulator;
+    @Inject
+    private Logger log;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        try {
+            ModifiedRepositoryDescriptor descriptor = P2RepositoryDataManipulator.createDescriptor(repository,
+                    repositoryKind, outputLocation, compress, xzCompress, keepNonXzIndexFiles);
+
+            String type = "repository";
+            if (repositoryKind != P2RepositoryKind.both) {
+                type = (repositoryKind.isArtifact() ? "artifact" : "metadata") + " repository";
+            }
+            URI location = descriptor.repository().getURL();
+            Path output = descriptor.outputLocation();
+            if (output != null) {
+                log.info("  Write modified " + type + " file(s) to: " + output);
+            } else {
+                log.info("  Modify " + type + " in-place at: " + location);
+            }
+            if (repositoryName != null) {
+                log.info("  Set name to: " + repositoryName);
+            }
+            if (!propertiesToRemove.isEmpty()) {
+                log.info("  Remove properties:" + String.join(", ", propertiesToRemove));
+            }
+            if (!propertiesToAdd.isEmpty()) {
+                log.info("  Add properties:");
+                propertiesToAdd.forEach((key, value) -> {
+                    log.info("   " + key + " = " + value);
+                });
+            }
+
+            manipulator.modifyRepositoryMetadata(descriptor, repositoryName, propertiesToRemove, propertiesToAdd);
+        } catch (IOException | URISyntaxException e) {
```

Comment:
`URISyntaxException` more feels like it should be a `MojoFailureException` (as it indicates a user problem).

Reply:
Changed it so that it's catched at it's origin and wrapped into a special `RepositoryLocationURISyntaxException`, which is then rethrown as cause of a `MojoFailureException`.